pkg/subctl/cmd/join.go

```diff
@@ -185,12 +189,45 @@ func joinSubmarinerCluster(config *rest.Config, subctlData *datafile.SubctlData)
 	clusterCIDR, err = getPodCIDR(clusterCIDR, networkDetails)
 	exitOnError("Error determining the pod CIDR", err)
 
+	status.Start("* Discovering multi cluster details")
+	globalNetworks := getGlobalNetworks(subctlData)
+	err = checkOverlappingServiceCidr(globalNetworks)
+	status.End(err == nil)
+	exitOnError("Error validating overlapping CLusterCIDRs", err)
+
 	status.Start("Deploying Submariner")
 	err = submarinercr.Ensure(config, OperatorNamespace, populateSubmarinerSpec(subctlData))
 	status.End(err == nil)
 	exitOnError("Error deploying Submariner", err)
 }
 
+func checkOverlappingServiceCidr(networks map[string]*globalnet.GlobalNetwork) error {
+	for k, v := range networks {
+		overlap, err := globalnet.IsOverlappingCIDR(v.ServiceCIDRs, serviceCIDR)
+		if err != nil {
+			return fmt.Errorf("unable to validate overlapping ServiceCIDR: %s", err)
+		}
+		if overlap && k != clusterID {
+			return fmt.Errorf("invalid CIDR: %s overlaps with cluster %s", serviceCIDR, k)
+		}
+	}
+	return nil
+}
+
+func getGlobalNetworks(subctlData *datafile.SubctlData) map[string]*globalnet.GlobalNetwork {
+
+	brokerConfig := subctlData.GetBrokerConfig()
+	brokerSubmClient, err := submarinerClientset.NewForConfig(brokerConfig)
+	exitOnError("Unable to setup submariner connection to broker cluster", err)
```

Comment:
If I understand it correctly, it means the user has passed an invalid/old broker-info.subm file. If it is so, shall we give a message on those lines?

Reply:
Will do. I refactored the code a bit but forgot to update this msg from a generic one.